ophyd/areadetector/filestore_mixins.py

```diff
@@ -241,25 +292,66 @@ def write_path_template(self, val):
         self._write_path_template = _ensure_trailing_slash(val)
 
     def stage(self):
-        self._point_counter = count()
         self._locked_key_list = False
         self._datum_uids.clear()
         super().stage()
-        if self.enable.get() and self._reg is None:
-            raise RuntimeError('The plugin {!r} is enabled, but '
-                               'the Registry (self._reg) is `None`')
+        # Subclasses will assemble resource_kwargs and call
+        # self._generate_resource(resource_kwrags)
+
+    def _generate_resource(self, resource_kwargs):
+        fn = PurePath(self._fn).relative_to(self.reg_root)
+        resource, self._datum_factory = resource_factory(
+            spec=self.filestore_spec,
+            root=str(self.reg_root),
+            resource_path=str(fn),
+            resource_kwargs=resource_kwargs,
+            path_semantics=self.path_semantics)
+
+        # If a Registry is set, we need to allow it to generate the uid for us.
+        if self._reg is not None:
+            logger.debug("Inserting resource with filename %s", self._fn)
+            # register_resource has accidentally different parameter names...
```

Comment:
thanks for adding that comment. at least we can remember which one we're aware of that we want to move away from :-D